File: rdds/3.ratings.py
from pyspark import SparkContext

sc = SparkContext("local[*]", "ratings")
rdd1 = sc.textFile("path")

rdd2 = rdd1.map(lambda x: x.split("\t")[2])

rdd3 = rdd2.map(lambda x : (x,1))

result = rdd3.reduceByKey(lambda x,y : (x+y)).collect()

for a in result:
    print(a)

File: rdds/3.ratings.scala
// count for 5 rating
// count for 4 rating
// count for 3 rating
// count for 2 rating
// count for 1 rating


object ratings extends App {

    val sc = new SparkContext(local[*], "ratings")
    val input = sc.textfile("filepath")

    val mapped = input.map(x =>  x.split("\t")(2))
    val rating = mapped.map( x => (x,1))

    val reducedratings = rating.reduceByKey( (x,y) => x+y)
    val result = reducedratings.collect

// instead of using map where we are using (x,1) we can use following also and the same result can be achieved
//no need of extra lines

    val result = mapped.countByValue  // this is an action and obviously is not an RDD it is a local variable

}